File: n3/3.kt
fun main() {
    print("Введите количество чисел и желаемые числа: ")
    val input = readLine()!!.split(" ")
    val amount = input[0].toInt() // кол-во чисел (не используется в алгоритме)
    val numbers = input.subList(1, input.size) // список чисел
    val odd = mutableListOf<Int>() // список для хранения количества нечетных цифр

    for (numStr in numbers) {
        var count = 0
        for (digitChar in numStr) {
            val digit = digitChar.toString().toInt()
            if (digit % 2 != 0) {
                count++
            }
        }
        odd.add(count)
    }

    println("Количество нечетных цифр в числах: $odd")
}

File: n3/3.ts
import * as readline from 'readline';
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});
rl.question('Введите числа через пробел: ', (input: string) => {
    const lst: string[] = input.trim().split(/\s+/);
    
    //Получаем только числа (игнорируем первый элемент если нужно)
    const numbers: string[] = lst.slice(1);
    
    //Массив для результатов
    const odd: number[] = [];
    let count: number = 0;

    //Подсчет нечетных цифр в каждом числе
    for (const num of numbers) {
        for (const char of num) {
            if (parseInt(char) % 2 === 1) {
                count++;
            }
        }
        odd.push(count);
        count = 0;
    }

    console.log('Количество нечетных цифр:', odd);
    rl.close();
});
